### ISSUE TITLE
How to support Clickhouse UDF

### ISSUE THREAD

**Issue by oikomi** [spiceai/spiceai#2840]
As shown in the error below, when I use some clickhouse UDFs (similar to groupArray), an error will be reported

Query Error Invalid function 'grouparray'.

What should I do, thank you

<img width="721" alt="image" src="https://github.com/user-attachments/assets/7b124ba1-d96d-455c-ba03-de0a28210d9d">

**Comment by phillipleblanc** [comment]
We currently only support the functions that come built into DataFusion i.e.: https://datafusion.apache.org/user-guide/sql/aggregate_functions.html

We don't directly pass the query to the underlying DB, so if in your case the underlying DB is Clickhouse - even though it supports that `grouparray` function, since DataFusion doesn't understand it, it can't parse the SQL query. This is a limitation that we can consider improving in the future. (i.e. add a stub implementation for functions that are supported in the underlying DB, and during federation the functions will be run directly on Clickhouse).

**Comment by lukekim** [comment]
@oikomi would love to learn more about your use-case so we can prioritize this feature. DM me on our Discord?

**Comment by oikomi** [comment]
> @oikomi would love to learn more about your use-case so we can prioritize this feature. DM me on our Discord?

Yes, these needs are obvious because we have been operating a product for many years, and a large number of historical queries are based on clickhouse. We are unlikely to modify the syntax of so many queries (involving hundreds of people), so we need to be compatible with clickhouse syntax. Thanks

**Comment by lukekim** [comment]
@oikomi I don't believe you ever reached out, would you DM me on Discord, and we can learn about your scenario here.
